letmeinfwd/src/firewall/nftables.rs

```diff
@@ -407,7 +482,19 @@ impl NftFirewall {
     async fn nftables_remove_leases(&mut self, conf: &Config, leases: &[Lease]) -> ah::Result<()> {
         if !leases.is_empty() {
             // Get the active ruleset from the kernel.
-            let ruleset = ListedRuleset::from_kernel(conf).await?;
+            let ruleset = match ListedRuleset::from_kernel(conf).await {
+                Ok(ruleset) => ruleset,
+                Err(e) => {
+                    // In debug mode, handle the error gracefully
+                    if conf.debug() {
```

Comment:
This breaks backward compatibility. Debug mode is not supposed to change behavior.
If there is a good reason to ignore some errors, these cases should be explicitly coded and documented. It shall be preferred to not be dependent on the configuration whether errors are ignored or not at this level.